### PR TITLE
improve design 

### PR DESCRIPTION
**Pull request by SURYAKANTSHARMA**
- Add base target for iOS 11
- improve api design according to discussion  https://youtu.be/Qq-aCVVrIAc

### CODE REVIEW

File: Example/TestCountryPickerFramework/ViewController.swift [hunk 1]
```diff
@@ -88,34 +88,36 @@ private extension ViewController {
...
+                countryController.favoriteCountriesLocaleIdentifiers = ["IN", "US"]
+
+            }) { [weak self] country in
                 
                 guard let self = self else { return }
-                
                 self.countryImageView.isHidden = false
                 self.countryImageView.image = country.flag
                 self.countryCodeButton.setTitle(country.dialingCode, for: .normal)
             }
```
Comment (Sharkesm) [inline Example/TestCountryPickerFramework/ViewController.swift]: Optional: May as well capture the variables instead 

```
{ [countryImageView, countryCodeButton] country in
       countryImageView.isHidden = false
       countryImageView.image = country.flag
       countryCodeButton.setTitle(country.dialingCode, for: .normal)
}
```

**Review by Sharkesm — CHANGES_REQUESTED** [review]

**Review by Sharkesm — CHANGES_REQUESTED** [review]
Also, the example project is corrupted. I can't seem to compile the app. Can you have a look into it, thanks

File: CountryPicker/CountryPicker/CountryPickerController.swift [hunk 2]
```diff
@@ -115,17 +122,7 @@ open class CountryPickerController: UIViewController {
...
-        
-        if #available(iOS 11.0, *) {
-            self.navigationItem.searchController = searchController
-        } else {
-            tableView.tableHeaderView = searchController.searchBar
-        }
-        
+        navigationItem.searchController = searchController
         definesPresentationContext = true
     }
```
[the remark this answers is not in the document]
Comment (SURYAKANTSHARMA) [reply CountryPicker/CountryPicker/CountryPickerController.swift:127]: didn't get you ? in lazy var we are constructing searchController and here we are setting navigationItem's property Or can you suggest code for better

**Comment by SURYAKANTSHARMA** [comment]
> thanks

sure between whats the issue it is working for me. Can you try `pod install`

**Pull request merged**